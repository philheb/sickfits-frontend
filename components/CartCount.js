import styled from 'styled-components';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

const Dot = styled.div`
  background: var(--red);
  color: white;
  border-radius: 50%;
  padding: 0.7rem;
  /* line-height: 2rem;
  min-width: 3rem; */
  margin-left: 1rem;
  font-feature-settings: 'tnum';
  font-variant-numeric: tabular-nums;
`;

const AnimationStyles = styled.span`
  position: relative;
  .count {
    display: block;
    position: relative;
    transition: transform 0.7s;
    backface-visibility: hidden;
  }
  .count-enter {
    transform: scale(4) rotateX(180deg);
  }
  .count-enter-active {
    transform: rotateX(0);
  }
  .count-exit {
    top: 0;
    position: absolute;
    transform: rotateX(0);
  }
  .count-exit-active {
    transform: scale(4) rotateX(180deg);
  }
`;

export default function CartCount({ count }) {
  return (
    <AnimationStyles>
      <TransitionGroup>
        <CSSTransition
          unmountOnExit
          classNames="count"
          className="count"
          key={count}
          timeout={{ enter: 700, exit: 700 }}
        >
          <Dot>{count}</Dot>
        </CSSTransition>
      </TransitionGroup>
    </AnimationStyles>
  );
}
