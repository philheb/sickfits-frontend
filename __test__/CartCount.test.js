import { render } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount />', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });
  it('Matches snapshot', () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });
  it('updates via props', () => {
    const { container } = render(<CartCount count={11} />);
    expect(container.textContent).toBe('11'); // Same thing as expect(container).toHaveTextContent('11)
  });
});
