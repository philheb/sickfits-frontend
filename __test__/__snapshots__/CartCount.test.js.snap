// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> Matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-xxvp4g-1 gaipoK"
  >
    <div>
      <div
        class="CartCount__Dot-xxvp4g-0 eJulvk count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;
